src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java

```diff
@@ -515,6 +522,66 @@ private void testChangeFeedbackRecipient() {
         assertTrue(feedbackEditPage.verifyVisibilityOptionsIsDisplayed(1));
     }
     
+    private void testVisibilityOptionsCorrespondToFeedbackPath() {
+        ______TS("Changing feedback path will enable/disable the corresponding visibility options");
+
+        feedbackEditPage = getFeedbackEditPage();
+
+        feedbackEditPage.clickQuestionEditForQuestion1();
+        feedbackEditPage.clickEditLabel(1);
+        feedbackEditPage.selectGiverToBe(FeedbackParticipantType.SELF, 1);
+        feedbackEditPage.selectRecipientToBe(FeedbackParticipantType.SELF, 1);
+
+        assertEnabledVisibilityOptionsIncludesOnly(Arrays.asList(
+                FeedbackParticipantType.RECEIVER, FeedbackParticipantType.OWN_TEAM_MEMBERS,
+                    FeedbackParticipantType.RECEIVER_TEAM_MEMBERS, FeedbackParticipantType.STUDENTS,
+                    FeedbackParticipantType.INSTRUCTORS),
+                1);
+
+        feedbackEditPage.selectGiverToBe(FeedbackParticipantType.SELF, 1);
+        feedbackEditPage.selectRecipientToBe(FeedbackParticipantType.INSTRUCTORS, 1);
+
+        assertEnabledVisibilityOptionsIncludesOnly(Arrays.asList(
+                FeedbackParticipantType.RECEIVER, FeedbackParticipantType.OWN_TEAM_MEMBERS,
+                    FeedbackParticipantType.STUDENTS, FeedbackParticipantType.INSTRUCTORS),
+                1);
+
+        feedbackEditPage.selectGiverToBe(FeedbackParticipantType.SELF, 1);
+        feedbackEditPage.selectRecipientToBe(FeedbackParticipantType.NONE, 1);
+
+        assertEnabledVisibilityOptionsIncludesOnly(Arrays.asList(
+                FeedbackParticipantType.OWN_TEAM_MEMBERS, FeedbackParticipantType.STUDENTS,
+                    FeedbackParticipantType.INSTRUCTORS),
+                1);
+
+        feedbackEditPage.selectGiverToBe(FeedbackParticipantType.INSTRUCTORS, 1);
+        feedbackEditPage.selectRecipientToBe(FeedbackParticipantType.NONE, 1);
+
+        assertEnabledVisibilityOptionsIncludesOnly(Arrays.asList(
+                FeedbackParticipantType.STUDENTS, FeedbackParticipantType.INSTRUCTORS), 1);
+    }
+
+    private void assertEnabledVisibilityOptionsIncludesOnly(List<FeedbackParticipantType> expectedTypes,
+                                                           int questionNumber) {
+        Set<String> expectedEnabledOptions = new HashSet<String>();
+        for (FeedbackParticipantType expectedType : expectedTypes) {
+            expectedEnabledOptions.add(expectedType.toString());
+        }
+
+        Set<String> actualEnableOptions = new HashSet<String>();
```

Comment:
I've nothing against using `Set`, but doesn't using `List` yield the same result?